### PR TITLE
Drop using django-smtp-ssl and a few minor settings updates

### LINKED ISSUE #426: Remove dependency on django-smtp-ssl in favor of Django's native SSL support
Django supports SMTP over SSL [for a while now](https://github.com/django/django/commit/59ebe39812858ba37e83ab0ee886f676980d472d). Just make sure to add `EMAIL_USE_SSL=True` to the settings.

### PR DESCRIPTION
**Pull request by jezdez**
This fixes #426

### CODE REVIEW

File: cabot/settings_utils.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import os
+
+def environ_get_list(names, default):
```
Comment (frankh) [inline cabot/settings_utils.py]: This should probably have a default for default, `default=None`

**Comment by frankh** [comment]
Thanks for this. Just need to fix that broken tests - probably by not requiring default on environ_get_list

File: cabot/settings_utils.py (hunk 1 above)
Comment (jezdez) [inline cabot/settings_utils.py]: Woops, yep, it should.

**Comment by jezdez** [comment]
@frankh The tests seem to fail now after the sqlalchemy dep was removed but the database result backend is still in use

**Comment by frankh** [comment]
oops... re-added it

**Pull request merged**